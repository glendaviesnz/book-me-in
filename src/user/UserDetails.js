import React from 'react';
import styled from 'styled-components';
import Menu, { MenuItem } from 'material-ui/Menu';

import { logout } from '../services/authentication';

const ProfileWrapper = styled.div`
    display: flex;
    align-items: center;
`
const UserPhoto = styled.img`
    width: 40px;
    height: 40px;
    border-radius: 50%;
    margin-left: 10px;
`

class UserDetails extends React.Component {
    state = {
        anchorEl: null,
    };

    handleClick = event => {
        this.setState({ anchorEl: event.currentTarget });
    };

    handleClose = () => {
        this.setState({ anchorEl: null });
    };
    logout = () => {
        this.handleClose();
        logout();
    }
    render() {
        const { anchorEl } = this.state;
        return (
            <div>
                <ProfileWrapper className="fadeIn" onClick={this.handleClick}>
                    <div>{this.props.name}</div>
                    <UserPhoto src={this.props.photoURL} />
                </ProfileWrapper>
                <Menu
                    id="simple-menu"
                    anchorEl={anchorEl}
                    open={Boolean(anchorEl)}
                    onClose={this.handleClose}
                >
                    <MenuItem onClick={this.logout}>Logout</MenuItem>
                </Menu>
            </div>
        );
    }

}

export default UserDetails;