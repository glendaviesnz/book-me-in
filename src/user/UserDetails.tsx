import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import { css } from 'emotion'
import * as React from 'react';
import { translate } from 'react-i18next';

import { logout } from '../auth/authentication.service';

const profileStyles = css`
  display: flex;
  align-items: center;
`;
const photoStyles = css`
  width: 40px;
  height: 40px;
  border-radius: 50%;
  margin-left: 10px;
`;

interface IUserDetailsProps {
   name: string;
   photoURL?: string;
   t: any;
}
class UserDetails extends React.Component<IUserDetailsProps, any> {
  public state = {
    anchorEl: undefined
  };

  public openMenu = (event: any) => {
    this.setState({ anchorEl: event.currentTarget });
  };

  public closeMenu = () => {
    this.setState({ anchorEl: undefined });
  };

  public logout = () => {
    this.closeMenu();
    logout();
  };

  public render() {
    const { anchorEl } = this.state;
    return (
      <div className={profileStyles} aria-owns={anchorEl ? 'simple-menu' : undefined}
      aria-haspopup="true" onClick={this.openMenu}>
        
          <div className="user-name">{this.props.name}</div>
          <img className={photoStyles} src={this.props.photoURL} />
        
        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.closeMenu}
        >
          <MenuItem onClick={this.logout}>{this.props.t('logout')}</MenuItem>
        </Menu>
      </div>
    );
  }
}
export default translate('common')(UserDetails);

export const UserDetailsUnWrapped = UserDetails;
