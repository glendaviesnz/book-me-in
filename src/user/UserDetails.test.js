import React from 'react';
import { shallow } from 'enzyme';

import UserDetails from './UserDetails';
it('renders without crashing', () => {
  const wrapper = shallow(<UserDetails />);
  expect(wrapper).toMatchSnapshot();
});
