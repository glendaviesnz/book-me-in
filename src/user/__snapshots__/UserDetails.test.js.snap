// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDetails />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.div
          className="fadeIn"
          onClick={[Function]}
>
          <div />
          <styled.img
                    src={undefined}
          />
</styled.div>,
        <WithStyles(Menu)
          anchorEl={null}
          id="simple-menu"
          onClose={[Function]}
          open={false}
>
          <WithStyles(MenuItem)
                    onClick={[Function]}
          >
                    Logout
          </WithStyles(MenuItem)>
</WithStyles(Menu)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div />,
            <styled.img
              src={undefined}
/>,
          ],
          "className": "fadeIn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorEl": null,
          "children": <WithStyles(MenuItem)
            onClick={[Function]}
>
            Logout
</WithStyles(MenuItem)>,
          "id": "simple-menu",
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Logout",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.div
            className="fadeIn"
            onClick={[Function]}
>
            <div />
            <styled.img
                        src={undefined}
            />
</styled.div>,
          <WithStyles(Menu)
            anchorEl={null}
            id="simple-menu"
            onClose={[Function]}
            open={false}
>
            <WithStyles(MenuItem)
                        onClick={[Function]}
            >
                        Logout
            </WithStyles(MenuItem)>
</WithStyles(Menu)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div />,
              <styled.img
                src={undefined}
/>,
            ],
            "className": "fadeIn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorEl": null,
            "children": <WithStyles(MenuItem)
              onClick={[Function]}
>
              Logout
</WithStyles(MenuItem)>,
            "id": "simple-menu",
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Logout",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Logout",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "snapshotSerializers": Array [
      "enzyme-to-json/serializer",
    ],
  },
}
`;
